src/gui/DetailsWidget.cpp

```diff
@@ -32,325 +32,291 @@
     , m_currentGroup(nullptr)
-    , m_timer(nullptr)
-    , m_attributesTabWidget(nullptr)
-    , m_attachmentsTabWidget(nullptr)
-    , m_autotypeTabWidget(nullptr)
+    , m_step(0)
+    , m_totpTimer(new QTimer(this))
     , m_selectedTabEntry(0)
     , m_selectedTabGroup(0)
 {
     m_ui->setupUi(this);
 
-    connect(parent, SIGNAL(pressedEntry(Entry*)), SLOT(getSelectedEntry(Entry*)));
-    connect(parent, SIGNAL(pressedGroup(Group*)), SLOT(getSelectedGroup(Group*)));
-    connect(parent, SIGNAL(currentModeChanged(DatabaseWidget::Mode)), SLOT(setDatabaseMode(DatabaseWidget::Mode)));
-
-    m_ui->totpButton->setIcon(filePath()->icon("actions", "chronometer"));
-    m_ui->closeButton->setIcon(filePath()->icon("actions", "dialog-close"));
+    // Entry
+    m_ui->entryTotpButton->setIcon(filePath()->icon("actions", "chronometer"));
+    m_ui->entryCloseButton->setIcon(filePath()->icon("actions", "dialog-close"));
 
-    connect(m_ui->totpButton, SIGNAL(toggled(bool)), SLOT(showTotp(bool)));
-    connect(m_ui->closeButton, SIGNAL(toggled(bool)), SLOT(hideDetails()));
-    connect(m_ui->tabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndex(int)));
+    m_ui->entryAttachmentsWidget->setReadOnly(true);
+    m_ui->entryAttachmentsWidget->setButtonsVisible(false);
 
-    m_ui->attachmentsWidget->setReadOnly(true);
-    m_ui->attachmentsWidget->setButtonsVisible(false);
+    connect(m_ui->entryTotpButton, SIGNAL(toggled(bool)), m_ui->entryTotpWidget, SLOT(setVisible(bool)));
+    connect(m_ui->entryCloseButton, SIGNAL(toggled(bool)), SLOT(hide()));
+    connect(m_ui->entryTabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndexes()), Qt::QueuedConnection);
 
-    m_attributesTabWidget = m_ui->tabWidget->widget(AttributesTab);
-    m_attachmentsTabWidget = m_ui->tabWidget->widget(AttachmentsTab);
-    m_autotypeTabWidget = m_ui->tabWidget->widget(AutotypeTab);
+    connect(m_totpTimer, SIGNAL(timeout()), this, SLOT(updateTotp()));
 
-    this->hide();
+    // Group
+    m_ui->groupCloseButton->setIcon(filePath()->icon("actions", "dialog-close"));
+    connect(m_ui->groupCloseButton, SIGNAL(toggled(bool)), SLOT(hide()));
+    connect(m_ui->groupTabWidget, SIGNAL(tabBarClicked(int)), SLOT(updateTabIndexes()), Qt::QueuedConnection);
 }
 
 DetailsWidget::~DetailsWidget()
 {
 }
 
-void DetailsWidget::getSelectedEntry(Entry* selectedEntry)
+void DetailsWidget::setEntry(Entry* selectedEntry)
 {
     if (!selectedEntry) {
-        hideDetails();
+        hide();
         return;
     }
 
     m_currentEntry = selectedEntry;
 
-    if (!config()->get("GUI/HideDetailsView").toBool()) {
-        this->show();
-    }
-
-    m_ui->stackedWidget->setCurrentIndex(EntryPreview);
+    updateEntryHeaderLine();
+    updateEntryTotp();
+    updateEntryGeneralTab();
+    updateEntryNotesTab();
+    updateEntryAttributesTab();
+    updateEntryAttachmentsTab();
+    updateEntryAutotypeTab();
+
+    setVisible(!config()->get("GUI/HideDetailsView").toBool());
```

Comment:
This shouldn't really be a setting with a checkbox IMHO. The widget has a close button right now, but it's extremely frustrating, because the panel comes back as soon as I select another entry. This setting should be set to false whenever I click the close button and instead of a checkbox in the application settings, there should be a little toolbar button at the bottom edge of the screen to restore a previously closed panel (effectively setting this back to true).

Reply:
There are currently 3 ways to hide the detailsView:
- Permanently in the settings (some user asked for this a while back since the option was available in KeePassX 0.4)
- Dragging down the details panel, you can restore it by dragging it up (the case you are describing, also available in KeePassX 0.4)
- Clicking the close button (hiding it temporary)

Reply:
The first two are completely unnecessary if the third one would do what I expect it to do.

Reply:
The thing missing it's an option in the menubar like KeePassX 0.4 below

![istantanea_2018-01-06_21-02-18_2](https://user-images.githubusercontent.com/686894/34643479-19afcaba-f325-11e7-8bf6-c51827da65cd.png)

Reply:
As I said: I find that totally unnecessary. It only adds additional complexity although it could be as simple as hiding and restoring it in place with the state being remembered. I wouldn't use KeePassX 0.4 as a guideline how things have to be.

Reply:
The dragging option is mandatory since we are using QSplitter.
The visibility of the splitter depends on your Qt theme really

You can't connect the close button to the splitter easily since you should have to remove the DetailsWidget from the QSplitter defined in DatabaseWidget, and then re-add it.

We can drop the first option (the permanent hide) in favor of the splitter, but every time you open KeePassXC the DetailsView will pop-up. A configuration option is needed to remember the splitter state

PS: I don't like having KeePassX 0.4 as a guideline neither but users asked about those options since they are familiar with them

Reply:
We have a config option. Use that for remembering the state and give the settings checkbox the sack.
QSplitter collapsing can be turned off. I don't have anything against it in general, but it shouldn't be the only way and if the panel is collapsed, a tool button should be shown for restoring it, just as it should be for closing with the X button. I imagine it like panels in IntelliJ/CLion with View->Tool Buttons enabled.

Reply:
@phoerious  @TheZ3ro guys can you describe what changes I need to make?

Reply:
I would prefer it as I described above. Remove the settings checkbox and instead use the setting to automatically remember the panel state. Add a tool button at the bottom to toggle panel visibility.

Reply:
It might be better to add a show/hide button to the toolbar? KeePassXC doesn't have as many panels as IntelliJ IDEA

Reply:
Maybe we can fix it in a follow-up PR when we decide a better method to hide it

Reply:
I can offer to add this to the 'View' menu (see PR #1305):

![screenshot_20180114_071144](https://user-images.githubusercontent.com/14027079/34913610-24813cb6-f902-11e7-90c1-6bc0fc59da55.png)